.github/workflows/ci-cd.yml

```diff
@@ -19,98 +19,106 @@ jobs:
-      with:
-        python-version: ${{ matrix.python-version }}
-        allow-prereleases: true
-    - name: Lock and sync dependencies
-      env:
-        AIOHTTP_VERSION: ${{ matrix.aiohttp-version }}
-      run: |
-        python -c "import sys; print(f'Python version: {sys.version}')"
-        python -m pip install -U setuptools pip codecov wheel pip-tools
-        time pip-compile requirements-dev.in
-        time pip-sync requirements-dev.txt
-    - name: Run pre-commit hooks
-      if: matrix.python-version == '3.11'
-      run: |
-        make pre-commit
-    - name: Run unittests
-      env:
-        COLOR: 'yes'
-      run: |
-        make mototest
-    - name: Upload coverage to Codecov
-      if: matrix.python-version == '3.11'
-      uses: codecov/codecov-action@v4.5.0
-      with:
-        token: ${{ secrets.CODECOV_TOKEN }} # not required for public repos
-        files: ./coverage.xml
-        flags: unittests # optional
-        name: codecov-umbrella # optional
-        fail_ci_if_error: true # optional (default = false)
-        verbose: true # optional (default = false)
-
-
-  pre-deploy:
-    name: Pre-Deploy
-    runs-on: ubuntu-latest
-    needs: test
-    # Run only on pushing a tag
-    if: github.event_name == 'push' && contains(github.ref, 'refs/tags/')
-    steps:
-      - name: Dummy
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+      - name: Setup Python ${{ matrix.pyver }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+          allow-prereleases: true
+      - name: Lock and sync dependencies
+        env:
+          AIOHTTP_VERSION: ${{ matrix.aiohttp-version }}
+        run: |
+          python -c "import sys; print(f'Python version: {sys.version}')"
+          python -m pip install -U pip codecov pip-tools
+          time pip-compile requirements-dev.in
+          time pip-sync requirements-dev.txt
+      - name: Run pre-commit hooks
+        if: matrix.python-version == '3.11'
+        run: |
+          make pre-commit
+      - name: Run unittests
+        env:
+          COLOR: 'yes'
         run: |
-            echo "Predeploy step"
+          make mototest
+      - name: Upload coverage to Codecov
+        if: matrix.python-version == '3.11'
+        uses: codecov/codecov-action@v4.5.0
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }} # not required for public repos
+          files: ./coverage.xml
+          flags: unittests # optional
+          name: codecov-umbrella # optional
+          fail_ci_if_error: true # optional (default = false)
+          verbose: true # optional (default = false)
 
-  build-tarball:
-    name: Tarball
+
+  # based on https://docs.github.com/en/actions/use-cases-and-examples/building-and-testing/building-and-testing-python#publishing-to-pypi
+  release-build:
     runs-on: ubuntu-latest
-    needs: pre-deploy
+
+    needs:
+      - test
```

Comment:
Don't depend on the tests. If anything, it's the tests that should depend on buildig the dists. This way you'd be able to test said dists instead of building things in those jobs again.

Reply:
I remove the dependency and opened #1169